packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -310,15 +316,18 @@ class GoTrueClient {
     await _asyncStorage!
         .removeItem(key: '${Constants.defaultStorageKey}-code-verifier');
 
-    final authResponse = AuthResponse.fromJson(response);
+    final authSessionUrlResponse = AuthSessionUrlResponse(
+        session: Session.fromJson(response)!, redirectType: redirectType?.name);
 
-    final session = authResponse.session;
-    if (session != null) {
```

Comment:
Got rid of this check for session. 

The signature of the return type on the js SDK guarantees that there is a session, so not sure why the js SDK has the check for session there. 
https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L534